src/Integration.Vsix/SonarLintTracker/IssueTracker.cs

```diff
@@ -128,13 +128,17 @@ private IssuesSnapshot TranslateMarkerSpans()
         internal void UpdateIssues(IEnumerable<Issue> issues)
         {
             var oldSnapshot = this.Factory.CurrentSnapshot;
-            var newMarkers = issues.Select(CreateIssueMarker);
+            var newMarkers = issues.Select(CreateIssueMarker).Where(m => m != null);
             var newSnapshot = new IssuesSnapshot(this.FilePath, oldSnapshot.VersionNumber + 1, newMarkers);
             SnapToNewSnapshot(newSnapshot);
         }
 
         private IssueMarker CreateIssueMarker(Issue issue)
         {
+            if (issue.StartLine < 1 || currentSnapshot.LineCount <= issue.EndLine - 1)
```

Comment:
The code is fine as it is, but I figured out that you could extract this if statement into something like:

    private bool IsValidIssue(Issue issue) => issue.StartLine < 1 || ...;

and use it like this:

    var newMarkers = issues.Where(IsValidIssue).Select(CreateIssueMarker);

Of course, my naming cannot be worse :)